### PR TITLE
Stop logo clashing when pinwheel animates

### PR DESCRIPTION
**Pull request by caitpj**
When hovering over the Airflow logo in the top left, the pinwheel part of the logo cuts across the text part of the logo. This is (subjectively) poor design. In order to stop this from happening, while keeping the pinwheel animation, this PR has:

- set text to transition slightly to the right when the animation activates.

I have also set the pinwheel animation back to infinite. In my opinion, having read the [discussion](https://github.com/apache/airflow/issues/34019) that resulted in the change, it was not the correct decision to switch the animation from infinite over hover to 1.5 seconds. Another change was made, to allow users with a 'prefers-reduced-motion' preference to not have the animation playing, was the correct change, and did not need a further change to the length of time the pinwheel animates for.

This is a design change, which is always subjective, so totally understand if review wants to bin this. Also sorry in advance for anything bad I've done; it's my 1st PR, I promise to do better next time. 

Video demonstrating change:
Before

https://github.com/apache/airflow/assets/595491/e8b3632e-85f1-4fd8-8609-c18121cc723e

After

https://github.com/apache/airflow/assets/595491/82091acc-9c2d-4523-b9ff-66cdbfd1e506

### CODE REVIEW

**Comment by potiuk** [comment]
I think what would help to evaluate this change (not by me - I am pretty far from UI) is if you add some recorded videos before and after. It might be that what you see is somewhat result of your browser configuration and showing what you mean would be good to avoid wrong assumptions. Also showing effect of your change would greatly help in explaining what you really achieved with it.

**Comment by caitpj** [comment]
> I think what would help to evaluate this change (not by me - I am pretty far from UI) is if you add some recorded videos before and after. It might be that what you see is somewhat result of your browser configuration and showing what you mean would be good to avoid wrong assumptions. Also showing effect of your change would greatly help in explaining what you really achieved with it.

Great idea, thank you. I've added a before and after video to the description above.

**Comment by potiuk** [comment]
> > I think what would help to evaluate this change (not by me - I am pretty far from UI) is if you add some recorded videos before and after. It might be that what you see is somewhat result of your browser configuration and showing what you mean would be good to avoid wrong assumptions. Also showing effect of your change would greatly help in explaining what you really achieved with it.
> 
> Great idea, thank you. I've added a before and after video to the description above.

Nice. I like it. I wonder what our UI team will say - but it looks rather cool.

**Review by potiuk — APPROVED** [review]

File: airflow/www/static/css/main.css [hunk 1]
```diff
@@ -105,7 +119,17 @@ div.container {
 @media (prefers-reduced-motion: no-preference) {
   .navbar-brand:hover .brand-logo-pinwheel {
     transform-origin: 17.66px 17.66px;
-    animation: pinSpin 1.5s linear;
+    animation: pinSpin 1.5s linear infinite;
```
Comment (pankajkoti) [inline airflow/www/static/css/main.css:122]:
```suggestion
    animation: pinSpin 1.5s linear;
```
We wish to not have an infinite spin. There was an issue https://github.com/apache/airflow/issues/34019 logged previously about this and hence we removed the infinite here https://github.com/apache/airflow/pull/34020

Comment (caitpj) [reply airflow/www/static/css/main.css:122]: I mention in the second paragraph of the description why I think removing infinite spin was not the right action to take away from issue #34019

Comment (pankajkoti) [reply airflow/www/static/css/main.css:122]: Okay, thanks, sorry missed reading that 😐

**Comment by jedcunningham** [comment]
Maybe just move "Airflow" over a bit so that doesn't need to start moving too? In the "after" video it feels a bit jarring imo.

**Comment by potiuk** [comment]
> Maybe just move "Airflow" over a bit so that doesn't need to start moving too? In the "after" video it feels a bit jarring imo.

Now we are talking design :D

**Comment by potiuk** [comment]
(and agree - when you move out the mouse, this is really disturbing when it suddenly goes back to start)

**Comment by ryanahamilton** [comment]
Figured I'd weigh in since I was the one who originally added this animation… The overlap was certainly something I noticed when implementing. I didn't increase the spacing between the word and pinwheel as I didn't want to alter the official brand logo, and I still don't believe we should. This was meant as a little Easter Egg interaction to add some fun to the UI. While I think this is a slick implementation, I think it is taking it a bit too far, and I'd advocate for removing the animation all together over making the animation more complex.

**Comment by bbovenzi** [comment]
I agree with Ryan, this was mainly an Easter Egg. I don't know if we want to mess with the logo too much. Can we double check everywhere else we use the Airflow SVG?

**Comment by caitpj** [comment]
> I agree with Ryan, this was mainly an Easter Egg. I don't know if we want to mess with the logo too much. Can we double check everywhere else we use the Airflow SVG?

@bbovenzi I've had a look at the code base again and there is no other use of the logo as an SVG other than this one (in navbar.html). All the other uses of the logo use a png.

**Comment by bbovenzi** [comment]
> > I agree with Ryan, this was mainly an Easter Egg. I don't know if we want to mess with the logo too much. Can we double check everywhere else we use the Airflow SVG?
> 
> @bbovenzi I've had a look at the code base again and there is no other use of the logo as an SVG other than this one (in navbar.html). All the other uses of the logo use a png.

Ok, let's just change the spacing between the pinwheel and the text then. I agree with Jed, I'm not a fan of the text shifting on hover.

**Comment by potiuk** [comment]
> > > I agree with Ryan, this was mainly an Easter Egg. I don't know if we want to mess with the logo too much. Can we double check everywhere else we use the Airflow SVG?
> > 
> > 
> > @bbovenzi I've had a look at the code base again and there is no other use of the logo as an SVG other than this one (in navbar.html). All the other uses of the logo use a png.
> 
> Ok, let's just change the spacing between the pinwheel and the text then. I agree with Jed, I'm not a fan of the text shifting on hover.

Damn... I knew I should never be a designer .

**Review by potiuk — CHANGES_REQUESTED** [review]
REquest changes so that it is not merged accidentally

**Comment by caitpj** [comment]
How does something like this look? The downside to this option is it doesn't give as tight a connection between the logo pinwheel and the logo text, saying that, it doesn't disconnect it too much in my opinion.
https://github.com/apache/airflow/assets/97813242/8f4f61d9-23ee-4553-a570-1ba758adb91a

**Comment by jscheffl** [comment]
> > > > I agree with Ryan, this was mainly an Easter Egg. I don't know if we want to mess with the logo too much. Can we double check everywhere else we use the Airflow SVG?
> > > 
> > > 
> > > @bbovenzi I've had a look at the code base again and there is no other use of the logo as an SVG other than this one (in navbar.html). All the other uses of the logo use a png.
> > 
> > 
> > Ok, let's just change the spacing between the pinwheel and the text then. I agree with Jed, I'm not a fan of the text shifting on hover.
> 
> Damn... I knew I should never be a designer .

Besides lagging the "Easter Egg" history I very much like the animation and would prefer to keep it. I am not a 100% designed but I treated the text shift as rather "cool" :-D But seems there are different opinions.

The commit https://github.com/apache/airflow/assets/97813242/8f4f61d9-23ee-4553-a570-1ba758adb91a also looks good for me though.

**Comment by eladkal** [comment]
> Besides lagging the "Easter Egg" history I very much like the animation and would prefer to keep it. I am not a 100% designed but I treated the text shift as rather "cool" :-D But seems there are different opinions.

 + 1

File: airflow/www/static/css/main.css [hunk 2]
```diff
@@ -105,7 +119,17 @@ div.container {
...
+  }
+
+  .navbar-brand:hover .brand-logo-divider {
+    width: 5px;
+    transition: width 0.5s;
+  }
+
+  .navbar-brand:hover .brand-logo-antidivider {
+    width: 0;
+    transition: width 0.5s;
```
Comment (bbovenzi) [inline airflow/www/static/css/main.css:132]: Is the width still animating? I'm happy with the tiny bit more spacing all the time.

**Pull request closed without merging**